3DAmsterdam/Assets/Amsterdam3D/Scripts/ObjectIDs/getBAGIDs.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using LayerSystem;
+
+    public class getBAGIDs : MonoBehaviour
+    {
+        public TileHandler tileHandler;
+        public GameObject BuildingContainer;
+        public bool isBusy = false;
+        private Ray ray;
+        private string id = "";
+        private GameObject selectedTile;
+        private bool mouseReleased = true;
+
+        private bool meshCollidersAttached = false;
+
+
+    // Update is called once per frame
+    void Update()
+    {
+
+        if (isBusy)
+        {
+
+            return;
+        }
+        if (!UnityEngine.EventSystems.EventSystem.current.IsPointerOverGameObject())
+        { 
+            if (Input.GetMouseButtonDown(0) == false)
+            {
+                mouseReleased = true;
+                return;
+            }
+            if (mouseReleased == true)
+            {
+                mouseReleased = false;
+
+                GetBagID();
+            }
+        }
+    }
+
+    private void GetBagID()
+    {
+        ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+        isBusy = true;
+        StartCoroutine(getIDData(ray, (value) => { UseObjectID(value); }));
+        DisplayBAGData.Instance.PrepareUI();
+    }
+
+    private void UseObjectID(string id)
+    {
+        if (id == "null")
+        {
+            BuildingContainer.GetComponent<LayerSystem.Layer>().UnHighlightAll();
+        }
+        else
+        {
+            BuildingContainer.GetComponent<LayerSystem.Layer>().Highlight(id);
+        }
+        isBusy = false;
+    }
+        IEnumerator LoadMeshColliders()
+        {
+            // add meshcolliders
+            MeshCollider meshCollider;
+            MeshFilter[] meshFilters = BuildingContainer.GetComponentsInChildren<MeshFilter>();
+        if (meshFilters == null)
+        {
+
```

Comment:
Gekke inspringing. Tip:  met visual studio kan de alle code selecteren en dan CTRL+K en dan CTRL+D doen om te auto formatten. Ezelsbruggetje voor gamers: (K)ill (D)eath

Reply:
Is opgelost